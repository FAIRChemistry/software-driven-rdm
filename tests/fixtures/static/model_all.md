# Test

This model tests all methods and types that are present within the sdRDM library.

## Objects

### Root

- str_value
  - Type: string
- float_value
  - Type: float
- int_value
  - Type: integer
- bool_value
  - Type: boolean
- date_value
  - Type: date
- datetime_value
  - Type: datetime
- posfloat_value
  - Type: posfloat
- posint_value
  - Type: PositiveInt
- http_url_value
  - Type: AnyHttpUrl
- email_value
  - Type: EmailStr
- bytes_value
  - Type: bytes
- multiple_primitives
  - Type: float
  - Multiple: True
- enum_value
  - Type: SomeEnum
- nested_single_obj
  - Type: Nested
- nested_multiple_obj
  - Type: Nested
  - Multiple: True
- referenced_value
  - Type: @Nested.id

### Nested

- str_value
  - Type: string
- float_value
  - Type: float
- int_value
  - Type: integer

## Enumerations

### SomeEnum

```python
VALUE1 = "value1"
VALUE2 = "value2"
```
