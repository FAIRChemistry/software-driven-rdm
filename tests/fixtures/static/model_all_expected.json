{
  "@context": {
    "Root": "file://tests/fixtures/static/model_all.md/Root"
  },
  "@type": ["Root"],
  "id": "id",
  "str_value": "string",
  "float_value": 1.5,
  "int_value": 1,
  "bool_value": true,
  "date_value": "2023-12-17",
  "datetime_value": "2023-12-17T15:55:44.245441",
  "posfloat_value": 1.5,
  "posint_value": 2,
  "http_url_value": "https://www.google.com/",
  "email_value": "max.mustermann@muster.de",
  "bytes_value": "this_is_bytes",
  "multiple_primitives": [1.5, 1.7, 1.9],
  "enum_value": "value1",
  "nested_single_obj": {
    "@context": {
      "Nested": "file://tests/fixtures/static/model_all.md/Nested"
    },
    "@type": ["Nested"],
    "id": "id",
    "str_value": "string",
    "float_value": 1.5,
    "int_value": 1
  },
  "nested_multiple_obj": [
    {
      "@context": {
        "Nested": "file://tests/fixtures/static/model_all.md/Nested"
      },
      "@type": ["Nested"],
      "id": "id",
      "str_value": "string",
      "float_value": 1.5,
      "int_value": 1
    }
  ],
  "leaf_element": {
    "@context": {
      "LeafElement": "file://tests/fixtures/static/model_all.md/LeafElement"
    },
    "@type": ["LeafElement"],
    "id": "id",
    "leaf_value": "I am a leaf",
    "some_attribute": "I am an attribute"
  }
}
